### DISCUSSION TITLE
Docker label based auto/dynamic configuration

### DISCUSSION THREAD

**Discussion by mxmeeple** [authelia/authelia#5023] [Ideas]
Since Authlia allows label configuration for almost everything except Access Policy (for understandable reasons), would it be possible to take the Traefik approach?
Where you add labels to a container/service and it'll will be detected and updated quickly without needing to restart Traefik

I've written a little helper container that reads container labels and spits out a config for me, using a shared volume to load it into Authelia. Its very clunky and would love to have a streamlined way of doing this
[authelia-traefik-docker-autoconfig](https://github.com/fenaren/authelia-traefik-docker-autoconfig)

**Comment by james-d-elliott** [comment]
While it's indeed possible to do this it's not entirely practical and fairly insecure for reasons to be discussed. 

Reasons:

1. Traefik is effectively special in this case, and we'd have to ensure the config is not tied to it in any way, thus the traefik labels would be ignored. In addition the rules would be hard to practically parse so if it was to use the traefik labels  it'd be insecure.
2. Many users have multiple rules for a single "container". For example you may want to secure a particular path more than others, or a particular method, etc. This cannot be well represented in a label. This combined with labels not having a particular order would result in either security or impracticality hell.
3. Rules are in a first match basis. So rules that come before are more impactful. Controlling this order is not entirely trivial.

**Comment by mxmeeple** [comment]
I think you've got slightly the wrong idea, I'm not suggesting using Traefik's labels, but a system similar to it. And not a single label either, but an array. eg:
```
--authelia.access_policy.portainer_web_access.domain=...
--authelia.access_policy.portainer_web_access.policy=one_factor
--authelia.access_policy.portainer_api_access.domain=...
``` 
1. It doesn't have to to use Traefik, its just a good example of a dynamic configuration system using docker labels
2. In the above example you can use different names in the 3rd section and define multiple paths that way
3. That is true, but an additional label could be used to specify priority 

On the security side I do appreciate that exposing docker socket is not ideal, but could be combined with socket proxies to reduce that risk
